src/lib.rs

```diff
@@ -1,7 +1,372 @@
+            self.shrink_to_fit();
+            Ok(())
+        }
+    }
+
+    /// Returns an iterator over the cache's key-value pairs in least- to
+    /// most-recently-used order.
+    pub fn iter(&self) -> Iter<K, V> {
+        Iter(self.inner.iter())
+    }
+
+    /// Returns a mutable iterator over the cache's key-value pairs in least- to
+    /// most-recently-used order.
+    pub fn iter_mut(&mut self) -> IterMut<K, V> {
+        IterMut(self.inner.iter_mut())
+    }
+    fn shrink_to_fit(&mut self) {
+        while self.current > self.max && !self.inner.is_empty() {
+            let (_, v) = self.inner.pop_front().expect("Not empty");
+            self.current -= V::measure(&v);
+        }
+    }
+}
+impl<K: Hash + Eq, V, S: BuildHasher + Default> IntoIterator for WeightCache<K, V, S> {
+    type Item = (K, V);
+    type IntoIter = IntoIter<K, V>;
+    fn into_iter(self) -> Self::IntoIter {
+        IntoIter(self.inner.into_iter())
+    }
+}
+/// An iterator over a cache's key-value pairs in least- to most-recently-used
+/// order.
+#[derive(Clone)]
+pub struct IntoIter<K, V>(linked_hash_map::IntoIter<K, V>);
+
+impl<K, V> Iterator for IntoIter<K, V> {
+    type Item = (K, V);
+
+    fn next(&mut self) -> Option<(K, V)> {
+        self.0.next()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.0.size_hint()
+    }
+}
+
+impl<K, V> DoubleEndedIterator for IntoIter<K, V> {
+    fn next_back(&mut self) -> Option<(K, V)> {
+        self.0.next_back()
+    }
+}
+
+impl<K, V> ExactSizeIterator for IntoIter<K, V> {
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+}
+/// An iterator over a cache's key-value pairs in least- to most-recently-used
+/// order.
+///
+/// Accessing a cache through the iterator does _not_ affect the cache's LRU state.
+pub struct Iter<'a, K: 'a, V: 'a>(linked_hash_map::Iter<'a, K, V>);
+
+impl<'a, K, V> Clone for Iter<'a, K, V> {
+    fn clone(&self) -> Iter<'a, K, V> {
+        Iter(self.0.clone())
+    }
+}
+
+impl<'a, K, V> Iterator for Iter<'a, K, V> {
+    type Item = (&'a K, &'a V);
+    fn next(&mut self) -> Option<(&'a K, &'a V)> {
+        self.0.next()
+    }
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.0.size_hint()
+    }
+}
+
+impl<'a, K, V> DoubleEndedIterator for Iter<'a, K, V> {
+    fn next_back(&mut self) -> Option<(&'a K, &'a V)> {
+        self.0.next_back()
+    }
+}
+
+impl<'a, K, V> ExactSizeIterator for Iter<'a, K, V> {
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+}
+/// An iterator over a cache's key-value pairs in least- to most-recently-used
+/// order with mutable references to the values.
+///
+/// Accessing a cache through the iterator does _not_ affect the cache's LRU state.
+pub struct IterMut<'a, K: 'a, V: 'a>(linked_hash_map::IterMut<'a, K, V>);
+
+impl<'a, K, V> Iterator for IterMut<'a, K, V> {
+    type Item = (&'a K, &'a mut V);
+    fn next(&mut self) -> Option<(&'a K, &'a mut V)> {
+        self.0.next()
```

Comment:
Is this safe? If you mutate the values and then remove, you could get some weird behaviour. E.g. you mutate small values to large values, but the weight of the cache does not get updated. So now you have a cache that exceeds its allocated size by a lot. Or am I missing something?

Reply:
You're right. I guess I will just remove `IterMut` completely.